avocado/core/task/statemachine.py

```diff
@@ -47,6 +47,69 @@ async def complete(self):
                            self._ready, self._started])
         return not pending
 
+    async def abort(self, status_reason=None, failfast=False):
+        """Abort all non-started tasks.
+
+        This method will move all non-started tasks to finished with a specific
+        reason.
+
+        :param status_reason: string reason. Optional.
+        :param failfast: Boolean that will determine if we should check for
+                         each task failfast config value. If this is True, it
+                         means the abort was caused by a 'failfast' event so we
+                         should abort only tasks that has failfast set to true.
+                         If is set to false (default).
+        """
+        await self.abort_queue('requested', status_reason, failfast)
+        await self.abort_queue('triaging', status_reason, failfast)
+        await self.abort_queue('ready', status_reason, failfast)
+
+    async def abort_queue(self, queue_name, status_reason=None,
+                          failfast=False):
+        """Abort all tasks inside a specific queue adding a status reason.
+
+        :param queue_name: a string with the queue name.
+        :param status_reason: string reason. Optional.
+        :param failfast: Boolean that will determine if we should check for
+                         each task failfast config value. If this is True, it
+                         means the abort was caused by a 'failfast' event so we
+                         should abort only tasks that has failfast set to true.
+                         If is set to false (default).
+        """
+        to_remove = []
+        async with self._lock:
+            queue = getattr(self, queue_name)
+            for _ in range(len(queue)):
+                if queue_name == 'requested':
+                    runtime_task = queue.popleft()
+                else:
+                    runtime_task = queue.pop(0)
+                enabled = runtime_task.task.runnable.config.get('run.failfast')
+                # Bring back to the queue when failfast (for the specific task)
+                # is false
+                if failfast and not enabled:
```

Comment:
Hi @beraldoleal I think this is not necessary, because IMO the jobs suites are not run in parallel and for each suite the new state machine instance is created. So if `failfast` is enabled, it will be enabled for all tasks in state machine. Am I right?

Reply:
You are absolutely right. ;) Good catch.

Reply:
@richtja fixed with a force-push.